ddtrace/internal/datadog/profiling/src/interface.cpp

```diff
@@ -0,0 +1,171 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
+// High-level skip for invalid architectures
+#ifndef __linux__
+#elif __aarch64__
+#elif __i386__
+#else
+
+#include "interface.hpp"
+#include "exporter.hpp"
+
+#include <iostream>
+#include <thread>
+
+// State
+bool is_initialized = false;
+Datadog::Uploader *g_uploader;
```

Comment:
Minor: I think we can wrap the c++ classes in cython objects
https://cython.readthedocs.io/en/latest/src/userguide/wrapping_CPlusPlus.html

Reply:
This is probably something we can do in a latter PR.

Reply:
Yup, indeed.  Doing so will be cleaner and generate better code.  I'd like to hold it off for a later PR, like you say, though.  👍